mm2src/mm2_main/tests/docker_tests/slp_tests.rs

```diff
@@ -147,3 +150,59 @@ fn test_withdraw_bch_max_must_not_spend_slp() {
 
     assert_eq!(expected_spendable, slp_balance.balance);
 }
+
+#[test]
+fn test_disable_platform_coin_with_tokens() {
+    let mm = slp_supplied_node();
+    let _ = block_on(enable_bch_with_tokens(
+        &mm,
+        "FORSLP",
+        &["ADEXSLP"],
+        UtxoRpcMode::Native,
+        false,
+    ));
+    // Try to disable ADEXSLP token.
+    assert_ok_200(&mm, "ADEXSLP", "disable_coin");
+    // Check if platform_coin FORSLP is still enabled.
+    assert_ok_200(&mm, "FORSLP", "my_balance");
+    // Check if ADEXSLP token still enabled.
+    assert_error_500(&mm, "ADEXSLP", "my_balance");
+    // // Try to disable patform_coin.
+    assert_ok_200(&mm, "FORSLP", "disable_coin");
+
+    // Enable enable_bch_with_tokens again to restart the process
+    let _ = block_on(enable_bch_with_tokens(
+        &mm,
+        "FORSLP",
+        &["ADEXSLP"],
+        UtxoRpcMode::Native,
+        false,
+    ));
+    // Try to disable platform_coin.
+    assert_ok_200(&mm, "FORSLP", "disable_coin");
+    // Check if platform_coin FORSLP is still enabled.
+    assert_error_500(&mm, "FORSLP", "my_balance");
+    // Check if ADEXSLP token is still enabled.
+    assert_error_500(&mm, "ADEXSLP", "my_balance");
+}
+
+fn assert_error_500(mm: &MarketMakerIt, token: &str, method: &str) {
+    let balance = block_on(mm.rpc(&json! ({
+        "userpass": mm.userpass,
+        "method": method,
+        "coin": token
+    })))
+    .unwrap();
+    assert_eq!(balance.0, StatusCode::INTERNAL_SERVER_ERROR);
+    assert!(balance.1.contains(&format!("No such coin: {token}")));
+}
+
+fn assert_ok_200(mm: &MarketMakerIt, token: &str, method: &str) {
```

Comment:
This function is also restricted to certain methods that only have `coin` field in them and no other fields. You can move it inside the test function too.